File: week1-python/Getting_started_with_images/Manipulating_Color_Pixels.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Path of the image to be loaded
imagePath = DATA_PATH + "/images/number_zero.jpg"
testImage = cv2.imread(imagePath,1)

#Access color pixel
print(testImage[0,0])

#Modify Pixel
testImage[0,0] = (0,255,255)
#cv2.imshow("Zero 1",testImage)
cv2.imwrite("../results/zero1.png",testImage)

testImage[1,1] = (255,255,0)
#cv2.imshow("Zero 2",testImage)
cv2.imwrite("../results/zero2.png",testImage)

testImage[2,2] = (255,0,255)
#cv2.imshow("Zero 3",testImage)
cv2.imwrite("../results/zero3.png",testImage)

#Modify Region of Interest
testImage[0:3,0:3] = (255,0,0)
testImage[3:6,0:3] = (0,255,0)
testImage[6:9,0:3] = (0,0,255)

#cv2.imshow("Zero", testImage)
#cv2.waitKey(0)
#cv2.destroyAllWindows()
cv2.imwrite("../results/zero.png",testImage)

File: week1-cpp/Getting_started_with_images/Manipulating_Color_Pixels.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include "dataPath.hpp"

using namespace std;
using namespace cv;

int main(void)
{
	string imagePath = DATA_PATH + "/images/number_zero.jpg";

	Mat testImage = imread(imagePath,1);

	cout << testImage.at<Vec3b>(0,0);

	testImage.at<Vec3b>(0,0) = Vec3b(0,255,255);
	imwrite("../results/zero1.png",testImage);
	//imshow("Zero 1",testImage);

	testImage.at<Vec3b>(1,1) = Vec3b(255,255,0);
	imwrite("../results/zero2.png",testImage);
	//imshow("Zero 2",testImage);

	testImage.at<Vec3b>(2,2) = Vec3b(255,0,255);
	imwrite("../results/zero3.png",testImage);
	//imshow("Zero 3",testImage);

	//waitKey(0);
	return 0;
}
